components/asana/asana.app.mjs

```diff
@@ -0,0 +1,429 @@
+import axios from "axios";
+
+export default {
+  type: "app",
+  app: "asana",
+  propDefinitions: {
+    organizations: {
+      label: "Organizations",
+      description: "List of organizations. This field use the organization GID.",
+      type: "string[]",
+      async options() {
+        const organizations = await this.getOrganizations();
+
+        return organizations.map((organization) => {
+          return {
+            label: organization.name,
+            value: organization.gid,
+          };
+        });
+      },
+    },
+    workspaces: {
+      label: "Workspaces",
+      description: "List of workspaces. This field use the workspace GID.",
+      type: "string[]",
+      async options() {
+        const workspaces = await this.getWorkspaces();
+
+        return workspaces.map((workspace) => {
+          return {
+            label: workspace.name,
+            value: workspace.gid,
+          };
+        });
+      },
+    },
+    teams: {
+      label: "Teams",
+      description: "List of teams. This field use the team GID.",
+      type: "string[]",
+      async options() {
+        const teams = await this.getTeams();
+
+        return teams.map((team) => {
+          return {
+            label: team.name,
+            value: team.gid,
+          };
+        });
+      },
+    },
+    projects: {
+      label: "Projects",
+      description: "List of projects. This field use the project GID.",
+      type: "string[]",
+      async options() {
+        const projects = await this.getProjects();
+
+        return projects.map((tag) => {
+          return {
+            label: tag.name,
+            value: tag.gid,
+          };
+        });
+      },
+    },
+    tags: {
+      label: "Tags",
+      description: "List of tags. This field use the tag GID.",
+      type: "string[]",
+      async options() {
+        const tags = await this.getTags();
+
+        return tags.map((tag) => {
+          return {
+            label: tag.name,
+            value: tag.gid,
+          };
+        });
+      },
+    },
+    users: {
+      label: "Users",
+      description: "List of users. This field use the user GID.",
+      type: "string[]",
+      async options() {
+        const users = await this.getUsers();
+
+        return users.map((user) => {
+          return {
+            label: user.name,
+            value: user.gid,
+          };
+        });
+      },
+    },
+    tasks: {
+      label: "Tasks",
+      description: "List of tasks. This field use the task GID.",
+      type: "string[]",
+      async options({ projects }) {
+        const tasks = await this.getTasks(projects);
+
+        return tasks.map((task) => {
+          return {
+            label: task.name,
+            value: task.gid,
+          };
+        });
```

Comment:
This can be simplified.
```suggestion
        return tasks.map((task) => ({
          label: task.name,
          value: task.gid,
        }));
```